### ISSUE TITLE
Ini config parsed incorrectly when using numeric indexes

### ISSUE THREAD

**Issue by hugofcampos** [phalcon/cphalcon#12725]
We are updgrade a Phalcon 1.* app and, in Phalcon 2.* to 3.0.4, our config files started to load incorrectly.

I'm trying to parse an ini array for logs config with this format:

```ini
channel.handlers.0.name = stream
channel.handlers.0.level = debug
channel.handlers.0.fingersCrossed = info
channel.handlers.0.filename = channel.log
channel.handlers.1.name = redis
channel.handlers.1.level = debug
channel.handlers.1.fingersCrossed = info
```

What I expected to see was:

```php
Array
(
    [channel] => Array
        (
            [handlers] => Array
                (
                    [0] => Array
                        (
                            [name] => stream
                            [level] => debug
                            [fingersCrossed] => info
                            [filename] => channel.log
                        )
                    [1] => Array
                        (
                            [name] => redis
                            [level] => debug
                            [fingersCrossed] => info
                        )
                )
        )
)
```

But this is what I got was:

```php
Array
(
    [channel] => Array
        (
            [handlers] => Array
                (
                    [0] => Array
                        (
                            [name] => stream
                        )

                    [1] => Array
                        (
                            [level] => debug
                        )

                    [2] => Array
                        (
                            [fingersCrossed] => info
                        )

                    [3] => Array
                        (
                            [filename] => channel.log
                        )

                    [4] => Array
                        (
                            [name] => redis
                        )

                    [5] => Array
                        (
                            [level] => debug
                        )

                    [6] => Array
                        (
                            [fingersCrossed] => info
                        )
                )

        )
    )
)
```

Apparently, integer pieces in string are not creating array indexes. This is a bug or an unsupported format?

```sh
# php --ri
phalcon


Web framework delivered as a C-extension for PHP
phalcon => enabled
Author => Phalcon Team and contributors
Version => 3.0.4
Build Date => Mar 20 2017 14:43:00
Powered by Zephir => Version 0.9.6a-dev-aef205594b

Directive => Local Value => Master Value
phalcon.db.escape_identifiers => On => On
phalcon.db.force_casting => Off => Off
phalcon.orm.events => On => On
phalcon.orm.virtual_foreign_keys => On => On
phalcon.orm.column_renaming => On => On
phalcon.orm.not_null_validations => On => On
phalcon.orm.exception_on_failed_save => Off => Off
phalcon.orm.enable_literals => On => On
phalcon.orm.late_state_binding => Off => Off
phalcon.orm.enable_implicit_joins => On => On
phalcon.orm.cast_on_hydrate => Off => Off
phalcon.orm.ignore_unknown_columns => Off => Off

# php -v
PHP 5.6.30 (cli) (built: Feb 28 2017 17:36:58)
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Xdebug v2.5.1, Copyright (c) 2002-2017, by Derick Rethans

# SO version
Debian GNU/Linux 8

Installation type: Compiling from source

Server: Nginx
```

**Comment by sergeyklay** [comment]
We'll try to sort out. As workaround you can use this approach:
```ini
[channel.handlers.0]
name = stream
level = debug
fingersCrossed = info
filename = channel.log
```

```
root@27712199a22a:/app# php -a
Interactive mode enabled

php > $config = new \Phalcon\Config\Adapter\Ini("/app/test.ini");
php > print_r($config->toArray());
Array
(
    [channel.handlers.0] => Array
        (
            [name] => stream
            [level] => debug
            [fingersCrossed] => info
            [filename] => channel.log
        )

)
```

**Comment by gguridi** [comment]
I would consider it an unsupported format.

The example can be done using associative arrays instead:

```php
channel.handlers.stream.name = stream
channel.handlers.stream.level = debug
channel.handlers.stream.fingersCrossed = info
channel.handlers.stream.filename = channel.log
channel.handlers.redis.name = redis
channel.handlers.redis.level = debug
channel.handlers.redis.fingersCrossed = info
```

Is there any reason why the handlers array should be an indexed array instead of associative? Other config formats support it anyway.

**Comment by hugofcampos** [comment]
It's possible, for sure. The numeric approach just seems more obvious for
us as we already have the "name" attribute and, as you said, is supported
to the other config formats.

Ps: we choose for change to php config files anyway, specially to make then
less limited by the format.

*Hugo Franco de Campos*

**Comment by sergeyklay** [comment]
actually you can try

`foo[] = bar`

**Comment by hugofcampos** [comment]
But I need a complex object for this index, not a scalar value.

**Comment by sergeyklay** [comment]
Fixed in the `3.4.x` branch. Feel free to open a new issue if the problem appears again. Thank you for contributing.
